src/ExperimentTable.js

```diff
@@ -137,29 +139,21 @@ class ExperimentTable extends React.Component {
     const displayedFields = tableHeader.map(header => header.dataParam)
     let experimentTableFilters=[]
 
-    tableFilters.map(
-      dropdown => {
-        const obj = {
-          label: ``,
-          options: []
-        }
-        obj.label = dropdown.label
-        aaData.filter(
-          data => Object.keys(data).map(
-            key => {
-              key === dropdown.dataParam ?
-                Array.isArray(data[key]) ?
-                  data[key].map(value =>
-                    !obj.options.includes(value) ?
-                      obj.options.push(value) : true) :
-                  !obj.options.includes((data[key])) ? obj.options.push(data[key]) : true : true
-              let index = experimentTableFilters.findIndex(label => label === dropdown.dataParam)
-              index !== -1 ?
-                experimentTableFilters[index].options = obj.options : experimentTableFilters.push(obj)
-            }))
-      })
-
-    experimentTableFilters = _.uniq(experimentTableFilters, filter => filter.options)
+    tableFilters.map(filter => {
+      const dropdownFilter = {
+        label: ``,
+        options: []
+      }
+      dropdownFilter.label = filter.label
+      aaData.filter(data => Object.keys(data).map(key => {
+        key === filter.dataParam ? Array.isArray(data[key]) ?
+          data[key].map(value => !dropdownFilter.options.includes(value) ? dropdownFilter.options.push(value) : true) :
+          !dropdownFilter.options.includes((data[key])) ? dropdownFilter.options.push(data[key]) : true : true
+      }))
+      let index = experimentTableFilters.findIndex(label => label === filter.dataParam)
+      index !== -1 ? experimentTableFilters[index].options = dropdownFilter.options :
+        experimentTableFilters.push(dropdownFilter)
+    })
```

Comment:
It is improved a bit, but still very confusing. Would you like to reply the questions I commented before of put comments for the logic>

Reply:
I want to create a structure for dropdowns like this:
`experimentTableFilters: [
{ label: ... ,
options: [...]
},
{...},
...
]`

I'm looping over the array of dropdowns being passed from props and creating an object of the structure `dropdownFilter: {
label: ...,
options: [...]
}` and then adding it in the array `experimentTableFilters` at the end of each loop.

To keep dropdown options distinct i'm checking if the option has already been added to the list `dropdownFilter.options`.

In the same way, I want to create one object per dropdown so before adding an `dropdownFilter` to `experimentTableFilters` I check if the `dropdownFilter` with same `label` exists.

Reply:
Thanks for the comments. Actually, I figured out a solution using `_.uniq` and `_.flatMap` to achieve what you want in two lines, but I will keep it secret for now. Would you like to have a go and we can discuss later?